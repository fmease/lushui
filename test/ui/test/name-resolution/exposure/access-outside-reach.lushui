;;; @Task more cases

module grotto of
    @public
    module deeper-end of
        @(public super) data Icicle: Type of

        @(public topmost)
        module rock-face of
            @(public topmost.grotto)
            data Ingrown-Stalactite: Type of

use grotto.deeper-end.((self as cave) (Icicle as I))
J: Type = cave.Icicle

module mod of
    use super.cave.rock-face.Ingrown-Stalactite as St
