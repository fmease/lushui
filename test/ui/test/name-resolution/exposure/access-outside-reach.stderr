error[E029]: binding ‘topmost.grotto.deeper-end.Icicle’ is private
   ┌─ ${TEST_FOLDER}/name-resolution/exposure/access-outside-reach.lushui:15:40
   │
15 │ use grotto.deeper-end.((self as cave) (Icicle as I))
   │                                        ══════

error[E029]: binding ‘topmost.grotto.deeper-end.Icicle’ is private
   ┌─ ${TEST_FOLDER}/name-resolution/exposure/access-outside-reach.lushui:16:16
   │
16 │ J: Type = cave.Icicle
   │                ══════

error[E029]: binding ‘topmost.grotto.deeper-end.rock-face.Ingrown-Stalactite’ is private
   ┌─ ${TEST_FOLDER}/name-resolution/exposure/access-outside-reach.lushui:19:30
   │
19 │     use super.cave.rock-face.Ingrown-Stalactite as St
   │                              ══════════════════

error: aborting due to 3 previous errors

